packages/pages/src/dev/server/middleware/types.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * The body of the Visual Editor preview POST request.
+ */
+export type VisualEditorPreviewOverrides = {
+  /** The site id. */
+  siteId: number;
+  /** The in-platform page set. */
+  pageSet: PageSet;
+  /** The dev mode Puck layout data. */
+  layout: Record<string, any>;
+  /** The dev mode theme data. */
+  theme: Record<string, any>;
+  /** The slug of the page that is being previewed. */
+  slug: string;
+  /** The external entityId of the page that is being previewed. */
+  entityId: string;
+  /** The locale of the page that is being previewed. */
+  locale: string;
+};
+
+/**
+ * The in-platform page set resource.
+ * Adapted from com/yext/publish/pagesets/resource.proto.
+ */
+export type PageSet = {
+  /** The AIP resource identifier. */
+  id: string;
+  /** The name of the code template the page set should use. */
+  codeTemplate: string;
+  /**
+   * Assignments define when to use this page set in new deployments for the site. They support the following:
+   * - GLOBAL to use this page set in all new deployments
+   * - PRODUCTION to use this page set in all new deployments for the production branch of the site
+   * - STAGING to use this page set in all new deployments for staging branches of the site
+   * - <branch name> to use this page set in all new deployments for the specified branch
+   * - <branch id> to use this page set in all new deployments for the specified branch
+   */
+  assignments: string[];
```

Comment:
Idk if we should model out the full thing. These will diverge over time. I think we just model out the things that are relevant to pagesjs.